mousestyles/behavior/construct_trees.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import print_function, absolute_import, division
+
+from .behavior_tree import BehaviorTree
+from mousestyles.behavior import metrics
+
+
+def process_raw_intervals(consumption, intervals, active_time,
+                          total_time, epsilon):
+    """
+    Takes `total_consumption` and the data in `intervals` and
+    decomposes it into a tree of behavioral summary statistics.
+
+    Parameters
+    ----------
+    consumption: float
+        total consumption (food, water, or movement) in the time period
+    intervals: Intervals
+        Intervals object representing intervals of behavior
+    epsilon: float
+        tolerance for merging events into bouts
+
+    Returns
+    -------
+    behavior : BehaviorTree
+        dictionary-like object representing breakdown of
+        consumption into various nodes
+
+    Examples
+    --------
+    >>> from mousestyles import data, intervals
+    >>> ints = intervals.Intervals(data.load_intervals('AS'))
+    >>> results = process_raw_intervals(1000, ints, 1)
+    """
```

Comment:
Make sure all the little calculations below are returning floats (not ints), so we don't get rounding errors